test/game/TestBoardState.java

```diff
@@ -215,10 +216,115 @@ public void testClone() {
 
 		BoardState b = new BoardState(pieces);
 		BoardState cloned = b.clone();
+		assertEquals(b, cloned);
 		b.movePiece(new Coordinate(0, 0), new Coordinate(1, 0));
-		
+
 		assertEquals(p1, cloned.getPieceAt(new Coordinate(0, 0)));
 		assertFalse(b.pieceAt(new Coordinate(0, 0)));
 		assertEquals(p1, b.getPieceAt(new Coordinate(1, 0)));
 	}
+
+	@Test
+	public void testEquals() {
+		HashMap<Coordinate, AbstractPiece> pieces1 = new HashMap<Coordinate, AbstractPiece>();
+		AbstractPiece p11 = new Cat(Owner.Player1);
+		AbstractPiece p12 = new Camel(Owner.Player1);
+		AbstractPiece p13 = new Horse(Owner.Player2);
+		AbstractPiece p14 = new Dog(Owner.Player2);
+
+		pieces1.put(new Coordinate(0, 0), p11);
+		pieces1.put(new Coordinate(1, 1), p12);
+		pieces1.put(new Coordinate(2, 2), p13);
+		pieces1.put(new Coordinate(3, 3), p14);
+
+		BoardState b1 = new BoardState(pieces1);
+
+		HashMap<Coordinate, AbstractPiece> pieces2 = new HashMap<Coordinate, AbstractPiece>();
+		AbstractPiece p21 = new Cat(Owner.Player1);
+		AbstractPiece p22 = new Camel(Owner.Player1);
+		AbstractPiece p23 = new Horse(Owner.Player2);
+		AbstractPiece p24 = new Dog(Owner.Player2);
+
+		pieces2.put(new Coordinate(0, 0), p21);
+		pieces2.put(new Coordinate(1, 1), p22);
+		pieces2.put(new Coordinate(2, 2), p23);
+		pieces2.put(new Coordinate(3, 3), p24);
+
+		BoardState b2 = new BoardState(pieces2);
+
+		HashMap<Coordinate, AbstractPiece> pieces3 = new HashMap<Coordinate, AbstractPiece>();
+		AbstractPiece p31 = new Cat(Owner.Player1);
+		AbstractPiece p32 = new Camel(Owner.Player1);
+		AbstractPiece p33 = new Horse(Owner.Player2);
+		AbstractPiece p34 = new Dog(Owner.Player2);
+
+		pieces3.put(new Coordinate(0, 0), p31);
+		pieces3.put(new Coordinate(1, 1), p32);
+		pieces3.put(new Coordinate(2, 2), p34);
+		pieces3.put(new Coordinate(3, 3), p33);
+
+		BoardState b3 = new BoardState(pieces3);
+
+		assertNotEquals(b1, null); // null test
+		assertEquals(b1, b1); // reflexive
+
+		assertEquals(b1, b2); // symmetric
+		assertEquals(b2, b1);
+
```

Comment:
not needed, because there are no subclasses, so they execute the same code, therefore, if its symmetric, its also transitive.